starlark/library.go

```diff
@@ -2183,6 +2188,74 @@ func set_union(_ *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error)
 	return union, nil
 }
 
+// https://github.com/google/starlark-go/blob/master/doc/spec.md#set·add.
+func set_add(_ *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error) {
+	var elem Value
+	if err := UnpackPositionalArgs(b.Name(), args, kwargs, 0, &elem); err != nil {
+		return nil, err
+	}
+	err := b.Receiver().(*Set).Insert(elem)
+	if err != nil {
+		return nil, nameErr(b, err)
+	}
+	return None, nil
+}
+
+// https://github.com/google/starlark-go/blob/master/doc/spec.md#set·remove.
+func set_remove(_ *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error) {
+	var k Value
+	if err := UnpackPositionalArgs(b.Name(), args, kwargs, 1, &k); err != nil {
+		return nil, err
+	}
+	if found, err := b.Receiver().(*Set).Delete(k); err != nil {
+		return nil, nameErr(b, err) // dict is frozen or key is unhashable
```

Comment:
If key is unhashable then it can't be a member of the set. Shouldn't it simply return in that case? I wonder what Python does.

Reply:
Good point - looks like Python 3.7.13 will throw a typeError when trying to remove (or discard) an unhashable type from a set.
```
>>> a = set([1,2,3])
>>> a.remove(1)
>>> a.remove([4,5,6])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'list'
```
 I think that this makes sense, especially in the case of `remove` where a successful return indicates that the item was found and removed. Thoughts?